src/components/schema-editor/schema-editor.component.tsx

```diff
@@ -2,49 +2,142 @@ import React from 'react';
+import { WarningFilled, CheckmarkFilled } from '@carbon/react/icons';
 import styles from './schema-editor.scss';
+import { useStandardFormSchema } from '../../hooks/useStandardFormSchema';
 
+interface markerProps {
+  startCol: number;
+  startRow: number;
+  endCol: number;
+  endRow: number;
+  className: string;
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  type: string | any;
+  text: string;
+}
 interface SchemaEditorProps {
   isLoading: boolean;
-  invalidJsonErrorMessage: string;
+  validation: boolean;
   onSchemaChange: (stringifiedSchema: string) => void;
   stringifiedSchema: string;
+  errors: Array<markerProps>;
+  setErrors: (errors: Array<markerProps>) => void;
 }
 
-const SchemaEditor: React.FC<SchemaEditorProps> = ({ invalidJsonErrorMessage, onSchemaChange, stringifiedSchema }) => {
-  const { t } = useTranslation();
+const SchemaEditor: React.FC<SchemaEditorProps> = ({
+  onSchemaChange,
+  stringifiedSchema,
+  errors,
+  setErrors,
+  validation,
+}) => {
+  const { schema } = useStandardFormSchema();
+
+  const validateJSON = (content: string, schema) => {
+    try {
+      const ajv = new Ajv({ allErrors: true, jsPropertySyntax: true });
+      const validate = ajv.compile(schema);
+      const parsedContent = JSON.parse(content);
+      const isValid = validate(parsedContent);
+
+      if (!isValid) {
+        const jsonLines = content.split('\n');
+        const errorMarkers = validate.errors.map((error) => {
+          let lineNumber = -1;
+          const instancePath = error.instancePath.replace(/^\./, '');
+          for (let i = 0; i < jsonLines.length; i++) {
+            if (jsonLines[i].includes(instancePath)) {
+              lineNumber = i;
+              break;
+            }
+          }
+          const message =
+            error.keyword === 'type'
+              ? `Invalid Type: ${error.instancePath.substring(1)} ${error.message}`
+              : `${capitalize(Object.keys(error.params)[0])}: ${
+                  error.message.charAt(0).toUpperCase() + error.message.slice(1)
+                }`;
+
+          return {
+            startRow: lineNumber,
+            startCol: 0,
+            endRow: lineNumber,
+            endCol: 1,
+            className: 'error',
+            text: message,
+            type: 'error',
+          };
+        });
+        setErrors(errorMarkers);
+      } else {
+        setErrors([]);
+      }
+    } catch (error) {
+      setErrors([]);
+    }
+  };
+
+  const debouncedValidateJSON = debounce(validateJSON, 300);
+
+  const handleChange = (newValue: string) => {
+    onSchemaChange(newValue);
+    debouncedValidateJSON(newValue, schema);
+  };
+
+  const ErrorMessages = () => (
+    <Toggletip className={styles.toggletipContainer} align="top-left">
+      <ToggletipButton label="Show information">
```

Comment:
This label should be run through the translation system.

Reply:
@denniskigen as if I remember Ian mentioning that carbon labels don't need to be translated, I don't think they will even be visible in the UI.